chatcommunicate.py

```diff
@@ -1,275 +1,355 @@
-
-            if result.message:
-                if wrap2.host + str(message_id) in GlobalVars.listen_to_these_if_edited:
-                    GlobalVars.listen_to_these_if_edited.remove(wrap2.host + str(message_id))
-                message_with_reply = u":{} {}".format(message_id, result.message)
-                if len(message_with_reply) <= 500 or "\n" in result.message:
-                    ev.message.reply(result.message, False)
-            if result.command_status is False:
-                add_to_listen_if_edited(wrap2.host, message_id)
-    except:
-        try:
-            # log the error
-            log_exception(*sys.exc_info())
-
-            ev.message.reply("I hit an error while trying to run that command; run `!!/errorlogs` for details.")
-            return
-        except:
-            print("An exception was thrown while handling an exception: ")
-            traceback.print_exc()
-
-
-# noinspection PyMissingTypeHints
-def handle_commands(content_lower, message_parts, ev_room, ev_room_name, ev_user_id, ev_user_name, wrap2, content,
-                    message_id):
-    message_url = "//chat.{host}/transcript/message/{id}#{id}".format(host=wrap2.host, id=message_id)
-    second_part_lower = "" if len(message_parts) < 2 else message_parts[1].lower()
-    if command_aliases.get(second_part_lower):
-        second_part_lower = command_aliases.get(second_part_lower)
-    match = re.match(r"[!/]*[\w-]+", content_lower)
-    command = match.group(0) if match else ""
-    if re.compile("^:[0-9]{4,}$").search(message_parts[0]):
-        msg_id = int(message_parts[0][1:])
-        msg = wrap2.get_message(msg_id)
-        msg_content = msg.content_source
-        quiet_action = ("-" in second_part_lower)
-        if str(msg.owner.id) != GlobalVars.smokeDetector_user_id[ev_room] or msg_content is None:
-            return Response(command_status=False, message=None)
-        post_url = fetch_post_url_from_msg_content(msg_content)
-        post_site_id = fetch_post_id_and_site_from_msg_content(msg_content)
-        if post_site_id is not None:
-            post_type = post_site_id[2]
+                processed_args.extend([get_type(arg) if arg else arg for get_type, arg in zip(type_signature, args)])
+
+                result = func(*processed_args, **({"alias_used": alias_used} if give_name else {}))
+                return result if not quiet_action else ""
+            except Exception as e:
+                return str(e)
+
+        cmd = (f, arity if arity else (len(type_signature), len(type_signature)))
+
+        if reply:
+            _commands["reply"][func.__name__] = cmd
+
+            for alias in aliases:
+                _commands["reply"][alias] = cmd
+        else:
+            _commands["prefix"][func.__name__] = cmd
+
+            for alias in aliases:
+                _commands["prefix"][alias] = cmd
+
+        return f
+
+    return decorator
```

Comment:
Could you check if the only argument passed to `@command` is a function, and if so, `return decorator(type_signature[0])`? This way you could

```py
@command
def foo(...):
  # ...
```

Reply:
Unfortunately the way the decorator works is that you pass in "types" which are really functions (int, float, str, message...) If you had something like @command(int) that would be indistinguishable from @command